Sources/Valet/SecureEnclave.swift

```diff
@@ -100,71 +55,74 @@ public final class SecureEnclave {
+    /// - Throws: An error of type `KeychainError`.
+    internal static func setObject(_ object: Data, forKey key: String, options: [String : AnyHashable]) throws {
         // Remove the key before trying to set it. This will prevent us from calling SecItemUpdate on an item stored on the Secure Enclave, which would cause iOS to prompt the user for authentication.
-        _ = Keychain.removeObject(forKey: key, options: options)
+        try Keychain.removeObject(forKey: key, options: options)
         
-        return Keychain.setObject(object, forKey: key, options: options).didSucceed
+        try Keychain.setObject(object, forKey: key, options: options)
     }
 
     /// - Parameters:
     ///   - key: A key used to retrieve the desired object from the keychain.
     ///   - userPrompt: The prompt displayed to the user in Apple's Face ID, Touch ID, or passcode entry UI.
     ///   - options: A base query used to scope the calls in the keychain.
-    /// - Returns: The data currently stored in the keychain for the provided key. Returns `.itemNotFound` if no object exists in the keychain for the specified key, or if the keychain is inaccessible. Returns `.userCancelled` if the user cancels the user-presence prompt.
-    internal static func object(forKey key: String, withPrompt userPrompt: String, options: [String : AnyHashable]) -> Result<Data> {
+    /// - Returns: The data currently stored in the keychain for the provided key.
+    /// - Throws: An error of type `KeychainError`.
+    internal static func object(forKey key: String, withPrompt userPrompt: String, options: [String : AnyHashable]) throws -> Data {
         var secItemQuery = options
         if !userPrompt.isEmpty {
             secItemQuery[kSecUseOperationPrompt as String] = userPrompt
         }
         
-        return Result(Keychain.object(forKey: key, options: secItemQuery))
+        return try Keychain.object(forKey: key, options: secItemQuery)
     }
 
     /// - Parameters:
     ///   - key: The key to look up in the keychain.
     ///   - options: A base query used to scope the calls in the keychain.
     /// - Returns: `true` if a value has been set for the given key, `false` otherwise.
-    internal static func containsObject(forKey key: String, options: [String : AnyHashable]) -> Bool {
+    /// - Throws: An error of type `KeychainError`.
+    internal static func containsObject(forKey key: String, options: [String : AnyHashable]) throws -> Bool {
         var secItemQuery = options
         secItemQuery[kSecUseAuthenticationUI as String] = kSecUseAuthenticationUIFail
-        
-        switch Keychain.containsObject(forKey: key, options: secItemQuery) {
-        case .success:
+
+        let status = Keychain.containsObject(forKey: key, options: secItemQuery)
+        switch status {
+        case errSecSuccess,
+             errSecInteractionNotAllowed:
```

Comment:
Might be worth adding a comment explaining why this means the item exists

Reply:
Resolved in 883ace0